### ISSUE TITLE
Make web UI more responsible with big number of torrents

### ISSUE THREAD

**Issue by antonsoroko** [elgatito/plugin.video.elementum#683]
## Expected Behavior
no significant lag in UI work with big number of torrents (>200)

## Current Behavior
with big number of torrent (>200) UI is freezing while it does updates, and since it does updates frequently, and look like update takes more time than update interval, UI is always laggy.

```
with 215 torrents
$ curl -o /dev/null http://192.168.31.30:65220/torrents/list
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 71465    0 71465    0     0   2142      0 --:--:--  0:00:33 --:--:-- 17366
```
with 175 torrents (but after restart)
```
$ curl -o /dev/null http://192.168.31.30:65220/torrents/list
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 71448    0 71448    0     0  11776      0 --:--:--  0:00:06 --:--:-- 14686
```

![Screenshot_20210124_154943](https://user-images.githubusercontent.com/17964763/105630962-eeb87400-5e5c-11eb-99ed-5dbc72ca4ffd.png)

update: but even with 90 torrents it is still significantly freezing.

## Possible Solution

if there is no easy way to speed up `list` request - probably a setting for update interval or dynamic update interval (based on number of torrents) will be enough.

## Steps to Reproduce (for bugs)

1. you need to have a lot of torrents (90 for example)+ file storage
2. open UI
3.
4.

## Your Environment

* Version used: 0.1.66

* Environment name and version: spmc 17.6

* Operating System and version: android tv 9

* Link to Kodi debug log:

https://paste.ubuntu.com/p/4ssgP3KmkN/

**Comment by elgatito** [comment]
@antonsoroko There is not source for WebUI, so this needs to be done from the very beginning.  
If someone makes it - I will include in the addon.

**Comment by elgatito** [comment]
@antonsoroko Oh, that is about backend response time. Please, post elementum's log bundle, there are recorded timers of functions executions.

**Comment by antonsoroko** [comment]
@elgatito 

https://yadi.sk/d/d2hGPVB9y-P8tg

debug/all + separate log (bundle does not work for smpc since spmc returns wrong log location)  + har file for browser dev panel

my guess - maybe some info (like number of peers) for one instance of torrent in `/list` is expensive to get so it became slow when you have a lot of torrents (i have 90 atm).


thank you!

**Comment by elgatito** [comment]
@antonsoroko Please, try with v0.1.67.
There is not much we can do, since each torrent needs information from libtorrent.

**Comment by antonsoroko** [comment]
@elgatito 

Thank you.
I compared network log to old one (i still have 91 torrents) - basically the same time (2.2s) and size (~37kB).
UI freezes for 2 seconds (get `\list`) and then unfreezes for about 2 seconds, so i guess update interval is about 5s.
With less than 100 torrents UI is annoying but still usable.

With information you provided - look like there is no easy way to fix this.
So i guess a note in readme than big amount of torrents will slow down the UI will be sufficient.

Thank you very much for looking into this issue!

**Comment by antonsoroko** [comment]
fixed by #740
